### ISSUE TITLE
Male displaced count show the negative results

### ISSUE THREAD

**Issue by felix-yew** [inasafe/inasafe#3973]
### Problem
I run the flood vector vs population vector and find a negative number of male displaced count in the detailed gender report.
![image](https://cloud.githubusercontent.com/assets/13621886/23448593/822676b6-fe84-11e6-8b95-152e28e4737c.png)

### Expected Behaviour
Show the right total displace for male.

### Environment
* InaSAFE : v.4.0.0b0
* QGIS : 2.14.10
* OS : Win 7
* Data used (filenames and paths so as to reproduce, a project is better):
  * Exposure : [DKI_Jakarta_Population_Dukcapil_2013_WGS84.zip](https://github.com/inasafe/inasafe/files/809869/DKI_Jakarta_Population_Dukcapil_2013_WGS84.zip)
  * Hazard : [Jakarta_Flood_18113_WGS84.zip](https://github.com/inasafe/inasafe/files/809872/Jakarta_Flood_18113_WGS84.zip)
  * Aggregation : -
  * Extent : 
  
Checklist (if applicable): 
* [x] Did you try to run it in debug mode ?
* [ ] Is the data available ?
  * If yes, where (BTSync, data.inasafe.org, zip) ? 

### Proposed Solution
Make it happy :smile: 

Cc: @Charlotte-Morgan @timlinux @Gustry @ismailsunni @lucernae @samnawi @ivanbusthomi

**Comment by Gustry** [comment]
@felix-yew Thanks for reporting this ticket and sharing your data.
But none of your zip (exposure and hazard) have a shapefile in (.shp).
I will try another dataset.

**Comment by felix-yew** [comment]
woops :pray:  sorry, I upload again
* Exposure: [DKI_Jakarta_Population_Dukcapil_2013_WGS84.zip](https://github.com/inasafe/inasafe/files/810297/DKI_Jakarta_Population_Dukcapil_2013_WGS84.zip)
* Hazard: [Jakarta_Flood_18113_WGS84.zip](https://github.com/inasafe/inasafe/files/810303/Jakarta_Flood_18113_WGS84.zip)
or you can check it in Real World Data in Flood folder.

let me know if there is some mistake in my zip file, again.

**Comment by Gustry** [comment]
I used the data from RealWorldData : 
I run the keywords wizard and everything runs fine for me : 
![screen shot 2017-03-01 at 13 48 17](https://cloud.githubusercontent.com/assets/1609292/23460467/d6b3f45a-fe85-11e6-98fe-dd5d9e63042c.png)

I used this scenario : 
```
[A Flood in Jakarta like 2013]
exposure = /Users/etienne/sync/RealWorldData/Flood/DKI Jakarta/02 Exposure Data/DKI_Jakarta_Population_Dukcapil_2013_WGS84.shp
hazard = /Users/etienne/sync/RealWorldData/Flood/DKI Jakarta/01 Hazard Data/Jakarta_Flood_18113_WGS84.shp
extent = 106.662524, -6.377878, 106.996513, -6.081991
extent_crs = EPSG:4326
```

Did you try to run the keyword wizard ?

**Comment by felix-yew** [comment]
Hi, @Gustry - I have set the keyword wizard of population data before I run the analysis, but I use the (female, youth, adult, and elderly) count. The count field attribute after the ratio field attribute.
![image](https://cloud.githubusercontent.com/assets/13621886/23489943/e5e97bb4-ff27-11e6-8da0-616e21922f3f.png)

I use the formula that have been explain by mas @samnawi in #3971 
![image](https://cloud.githubusercontent.com/assets/13621886/23490071/ab986d98-ff28-11e6-8212-72341c45d4ba.png)

I set the population data keyword
![image](https://cloud.githubusercontent.com/assets/13621886/23489656/46d177bc-ff26-11e6-95a9-6966314a7880.png)
And then, the result of male displaced count show the negative result.

I have updated the dukcapil population data in Real World Data.

**Comment by ismailsunni** [comment]
I can replicate it. I will work on it @felix-yew @Gustry 
<img width="1428" alt="screen shot 2017-03-02 at 14 50 09" src="https://cloud.githubusercontent.com/assets/1421861/23497847/94d32fce-ff57-11e6-9529-5a99de9bed62.png">

**Comment by felix-yew** [comment]
@Gustry @ismailsunni I think this issue was fixed in #3984 
Thank you

**Comment by ismailsunni** [comment]
Closed in #3984
